### PR TITLE
Added feature to change input for every request to avoid caching by jedi in python autocompletion

### PR DESCRIPTION
**Pull request by mansimarkaur**
@abhishekgahlot

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -5,23 +5,48 @@ import (
...
+	"sort"
 )
 
-var msg = []byte(`msg payload`)
+var msg = []byte(`{"body":{"code":"i","fileType":"python","line":0,"column":1,"wordToComplete":"i","offset":2}}`)
 var count = 1000
 var httpPort = 8900
+var lines []string
+
+func readLines() ([]string) {
```
Comment (abhishekgahlot) [inline cmd/main.go]: can you move this function someplace else not in main.go

File: cmd/main.go [hunk 2]
```diff
@@ -5,23 +5,48 @@ import (
...
 	runtime.GOMAXPROCS(runtime.NumCPU())
 	latency := make(chan []float64)
 	timeSeries := make(chan []time.Time)
 
-	currentTest := &carrot.Base{"example.com", "wss", 6000, msg, 100, 30, "/somepath"}
-	carrot.LoadTest(currentTest, latency, timeSeries)
+	lines := readLines()
+
+	currentTest := &carrot.Base{"localhost:8000", "ws", 100, msg, 10, 10, "/"}
+	carrot.LoadTest(currentTest, latency, timeSeries, lines)
```
Comment (abhishekgahlot) [inline cmd/main.go]: Don't pass big slice directly, pass as pointer or through channel. Slices are passed by value and may cause memory issue.

File: src/carrot/core.go [hunk 3]
```diff
@@ -4,6 +4,7 @@ import (
 	"log"
 	"sync"
 	"time"
+	"fmt"
```
Comment (abhishekgahlot) [inline src/carrot/core.go]: Also remove fmt here, its slow

File: src/carrot/core.go [hunk 4]
```diff
@@ -14,6 +15,7 @@ func receiveMsg(wsconn *websocket.Conn, done chan *Routine, rout *Routine) {
 		rout.ReceiveTime = time.Now()
 		rout.Diff = rout.ReceiveTime.Sub(rout.SendTime)
 		rout.ReceivedMsg = string(message)
+		fmt.Println(rout.ReceivedMsg)
```
Comment (abhishekgahlot) [inline src/carrot/core.go]: same here.

File: src/carrot/core.go [hunk 5]
```diff
@@ -22,24 +24,24 @@ func receiveMsg(wsconn *websocket.Conn, done chan *Routine, rout *Routine) {
 	}
 }
 
-func writeMsg(wsconn *websocket.Conn, base *Base, rout *Routine) {
+func writeMsg(wsconn *websocket.Conn, base *Base, rout *Routine, counter *Counter, line string) {
```
Comment (abhishekgahlot) [inline src/carrot/core.go:27]: use this line as part of base struct,

File: src/carrot/core.go [hunk 6]
```diff
@@ -22,24 +24,24 @@ func receiveMsg(wsconn *websocket.Conn, done chan *Routine, rout *Routine) {
...
 		return
 	}
-	go writeMsg(conn, base, rout)
+	go writeMsg(conn, base, rout, counter, line)
 	go receiveMsg(conn, doneCh, rout)
 	queue <- <-doneCh
 }
 
-func LoadTest(base *Base, latencyCh chan []float64, timeCh chan []time.Time) {
+func LoadTest(base *Base, latencyCh chan []float64, timeCh chan []time.Time, lines []string) {
```
Comment (abhishekgahlot) [inline src/carrot/core.go:63]: use pointers or simply use it from base struct, this is passed by value.

File: src/carrot/core.go [hunk 7]
```diff
@@ -50,7 +52,7 @@ func LoadTest(base *Base, latencyCh chan []float64, timeCh chan []time.Time) {
 
 	for range time.Tick(time.Millisecond * time.Duration(base.TickDelay)) {
 		routine := &Routine{time.Now(), time.Now(), 0, ""}
-		go singleTest(globalCounter, queue, base, routine)
+		go singleTest(globalCounter, queue, base, routine, lines[localCounter])
```
Comment (abhishekgahlot) [inline src/carrot/core.go:74]: Generally go shouldn't have data being passed to goroutines like this, you use channels for this thing.

File: src/carrot/structure.go [hunk 8]
```diff
@@ -13,6 +14,10 @@ type Base struct {
 	Path       string
 }
 
+func GenMsg(line string) []byte {
+	return []byte(fmt.Sprintf("%s%s%s%d%s%s%s%d%s", `{"body":{"code":"`, line, `","fileType":"python","line":0,"column":`, len([]rune(line)),`,"wordToComplete":"`, line, `","offset":`, len([]rune(line))+1, `}}`))
```
Comment (abhishekgahlot) [inline src/carrot/structure.go]: can we not use fmt or something.

**Comment by abhishekgahlot** [comment]
@mansimarkaur when you have time fix these or let it be like that, The suggested changes are important if you are contributing to carrot.

File: cmd/main.go (hunk 2 above)
Comment (mansimarkaur) [reply cmd/main.go]: Aren't slices just references to underlying arrays? Also, I found this: https://stackoverflow.com/questions/2439453/using-a-pointer-to-array
What am I missing here?

**Pull request closed without merging**